src/persistence/index.js

```
@@ -76,6 +76,105 @@ export class PersistDB {
+            contents: newContents = [],
+        } = newDB
+
+        if (newName !== DB_NAME) {
+            throw new Error(`Provided name "${newName}" does not match database name "${DB_NAME}".`)
+        }
+        if (Number(newVersion) !== DB_VERSION) {
+            throw new Error(`Provided version "${newVersion}" does not match database version "${DB_VERSION}."`)
+        }
+
+        PersistDB.open().then(async db => {
+            const tx = db.database.transaction(
+                ['states', 'changes', 'contents'], 'readwrite')
+            const states = tx.objectStore('states')
+            const changes = tx.objectStore('changes')
+            const contents = tx.objectStore('contents')
+
+            if (ids.length) {
+                let promises = []
+                ids.map(id => {
+                    promises.push(promisify(states.delete(id)))
+                    promises.push(promisify(contents.delete(id)))
+                })
+                await Promise.all([
+                    ...promises,
+                    iterate(changes, async (cursor, value) => {
+                        if (ids.includes(value.document)) {
+                            cursor.delete()
+                        }
+                        cursor.continue()
+                    }),
+                ])
+            } else {
+                await Promise.all([
+                    states.clear(),
+                    changes.clear(),
+                    contents.clear(),
+                ].map(promisify))
+            }
+
+            await Promise.all([newStates.map(st => {
+                return promisify(states.put(st))
+            })])
+
+            await Promise.all([newChanges.map(ch => {
+                return promisify(changes.put(ch))
+            })])
+
+            await Promise.all([newContents.map(ct => {
+                return promisify(contents.put(ct))
+            })])
+        }).catch(e => {
+            throw new Error(e)
+        })
```

Since this promise isn't returned, this will cause all errors to be silently ignored.